Duplicati/Library/Main/Operation/Backup/BackendUploader.cs

```diff
@@ -166,21 +172,20 @@ public Task Run()
                         }
                         else if (req is FlushRequest flush)
                         {
-                            try
+                            while (workers.Any())
                             {
-                                while (workers.Any())
+                                Task finishedTask = await Task.WhenAny(workers.Select(w => w.Task)).ConfigureAwait(false);
+                                if (finishedTask.IsFaulted)
                                 {
-                                    var finishedTask = await Task.WhenAny(workers.Select(w => w.Task)).ConfigureAwait(false);
-                                    if (finishedTask.IsFaulted)
-                                        ExceptionDispatchInfo.Capture(finishedTask.Exception).Throw();
-                                    workers.RemoveAll(w => w.Task == finishedTask);
+                                    flush.SetCanceled();
+                                    ExceptionDispatchInfo.Capture(finishedTask.Exception).Throw();
                                 }
-                                uploadsInProgress = 0;
-                            }
-                            finally
-                            {
+
+                                workers.RemoveAll(w => w.Task == finishedTask);
                                 flush.SetFlushed(lastSize);

```

Comment:
I think this should be outside of the while loop. Without tracking the task that corresponds to the `lastSize` variable we can't know when it is done.

Reply:
Nice catch, thanks.  Fixed in f52217e992c4738e93d10cc46a2aee6babba1c6f.